### PR TITLE
Update format for registry database

### PR DESCRIPTION
**Pull request by daniel-thom**

### CODE REVIEW

File: test_efs/dsgrid_project/datasets/modeled/comstock/dataset_with_dimension_ids.json5 [hunk 1]
```diff
@@ -32,42 +32,42 @@
   },
   dimension_references: [
     {
-      dimension_id: "us_counties_2010_-_comstock_only__b913be60-d3da-407b-935b-6a1b8ee7d92c",
```
Comment (daniel-thom) [inline test_efs/dsgrid_project/datasets/modeled/comstock/dataset_with_dimension_ids.json5:35]: This is an important change to call out. In the old way of doing things we could handle regenerating a registry by parsing these strings to separate dimension names and UUIDs, and then update the value with what is in the new registry. That isn't possible with the registry database, which stores dimension IDs as integers, and so I changed this format to use `name` explicitly. In order to use this file you have to run a script that converts it.

File: test_efs/dsgrid_project/datasets/modeled/comstock/dimension_mapping_config_with_ids.json5 [hunk 2]
```diff
@@ -5,12 +5,12 @@
       description: "Maps EFS ComStock prefix-formatted subsectors to the EFS project subsectors",
       file: "dimension_mappings/lookup_comstock_subsectors_to_project_subsectors.csv",
       from_dimension: {
-        dimension_id: "comstock_subsectors_efs_with_formatted_prefix__91f0b409-394b-4fba-9be8-3d4d74b2bcf8",
```
Comment (daniel-thom) [inline test_efs/dsgrid_project/datasets/modeled/comstock/dimension_mapping_config_with_ids.json5:8]: I did something very similar to dimension mappings.

File: test_efs/dsgrid_project/datasets/modeled/comstock/dimension_mapping_references.json5 [hunk 3]
```diff
@@ -3,20 +3,20 @@
     {
       from_dimension_type: "subsector",
       to_dimension_type: "subsector",
-      mapping_id: "comstock_subsectors_efs_with_formatted_prefix__comstock_subsectors_efs__0a072772-5072-4c98-8759-fa3124aa1cc9",
```
Comment (daniel-thom) [inline test_efs/dsgrid_project/datasets/modeled/comstock/dimension_mapping_references.json5:6]: And dimension mapping references

File: filtered_registries/simple_standard_scenarios/dump/projects_35a2864c314e4367eb1684bfe126a967.data.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...tor","mapping_id":"84141","version":"1.0.0","required_for_validation":true},{"from_dimension_type":"subsector","to_dimension_type":"subsector","mapping_id":"84165","version":"1.0.0","required_for_validation":true},{"from_dimension_type":"metric","to_dimension_type":"metric","mapping_id":"84189","version":"1.0.0","required_for_validation":true},{"from_dimension_type":"geography","to_dimension_type":"geography","mapping_id":"84213","version":"1.0.0","required_for_validation":true},{"from_dimension_type":"sector","to_dimension_type":"sector","mapping_id":"84237","version":"1.0.0","required_for_validation":true},{"from_dimension_type":"subsector","to_dimension_type":"subsector","mapping_id":"84261","version":"1.0.0","required_for_validation":true},{"from_dimension_type":"weather_year","to_dimension_type":"weather_year","mapping_id":"84285","version":"1.0.0","required_for_validation":true},{"from_dimension_type":"model_year","to_dimension_type":"model_year","mapping_id":"84309","version":"1.0.0","required_for_validation":true},{"from_dimension_type":"scenario","to_dimension_type":"scenario","mapping_id":"84333","version":"1.0.0","required_for_validation":true}],"dataset_to_project":{}}}
```
Comment (daniel-thom) [inline filtered_registries/simple_standard_scenarios/dump/projects_35a2864c314e4367eb1684bfe126a967.data.json:1]: This is an example of how ArangoDB exports data.

**Comment by daniel-thom** [comment]
@elainethale @lixiliu Can one of you approve this PR please?

**Review by elainethale — APPROVED** [review]

**Pull request merged**